xbmc/utils/EGLUtils.h

```diff
@@ -153,6 +158,11 @@ class CEGLAttributes
     return m_attributes.data();
   }
 
+  int Size() const
+  {
+    return m_writePosition - 1;
```

Comment:
```suggestion
    return m_writePosition;
```

I guess this also applies to `CEGLAttributesVec` - why would you subtract one for the size? It would return -1 for an empty array.

Reply:
because the the array always ends with `EGL_NONE`

Reply:
Ah, you are correct for `CEGLAttributesVec`, but not here. `m_writePosition` points *to* the `EGL_NONE` element, not *beyond*, so you do not have to subtract one.

Reply:
but I don't want to include `EGL_NONE`

Reply:
It does not. Just consider the default configuration after construction: `m_writePosition` is 0. The array data is [`EGL_NONE`, ... (undefined)]. Therefore, `Size()` should return `0`, which is `m_writePosition`.